### PR TITLE
1.0 release changes

### PR DESCRIPTION
**Pull request by adamyanalunas**
Going against any reasonable amount of restraint or focus I present the 1.0 candidate PR for loudspeaker.

Logic changes, architectural changes, properties and methods removed, properties and methods added, features added, accessibility enabled, new image assets; this PR has just about every type of change that should be in individual PRs—but all in one!

Good luck. 🎉

### CODE REVIEW

File: Example/Tests/LSPProgressViewSpec.m [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    it(@"is configured to tell iOS its accessibility information updates frequently", ^{
+        BOOL hasTraits = (view.progressBar.accessibilityTraits & UIAccessibilityTraitUpdatesFrequently) != 0;
+        expect(hasTraits).to.beTruthy();
+    });
+});
+
+
+SpecEnd
+
+//@interface LSPProgressViewSpecHelper : NSObject
```
Comment (SergeyKuryanov) [inline Example/Tests/LSPProgressViewSpec.m]: Leftover?

File: Example/loudspeaker/LSPViewController.m [hunk 2]
```diff
@@ -36,6 +38,13 @@ - (IBAction)launchDemo:(id)sender
 }
 
 
+- (void)closeDemo
```
Comment (SergeyKuryanov) [inline Example/loudspeaker/LSPViewController.m]: Bit confusing name, as method don't close anything.

File: Example/loudspeaker/LSPViewController.m [hunk 3]
```diff
@@ -46,100 +55,55 @@ - (void)configureDemo:(id)sender
 {
     if (sender == self.harlequinButton)
     {
-        self.audioVC.view.progressView.foregroundColor = UIColor.redColor;
-        self.audioVC.view.progressView.backgroundColor = UIColor.blueColor;
-        self.audioVC.view.backgroundColor = UIColor.greenColor;
-        self.audioVC.view.playbackTimeLabel.textColor = UIColor.brownColor;
-        self.audioVC.view.titleLabel.textColor = UIColor.yellowColor;
+        self.configuration = [LSPConfigurationBuilder configurationWithBuilder:^(LSPConfigurationBuilder *builder) {
+            builder.height = 120;
```
Comment (SergeyKuryanov) [inline Example/loudspeaker/LSPViewController.m]: Why do we need set height in configuration? Why not let users specify player frame?

File: Pod/Classes/LSPAudioPlayerModel.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+//
+//  LSPAudioPlayerModel.h
+//  Expecta
```
Comment (SergeyKuryanov) [inline Pod/Classes/LSPAudioPlayerModel.h]: What does Expecta mean?

File: Pod/Classes/LSPAudioViewController.m [hunk 5]
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
...
+        
+        [strongSelf.playerView mas_updateConstraints:^(MASConstraintMaker *make) {
+            make.bottom.equalTo(self.bottomLayout).with.offset(0);
+        }];
+        [strongSelf.view layoutIfNeeded];
+    } completion:nil];
+}
+
+
+- (void)hide:(nullable void (^)(void))completion
```
Comment (SergeyKuryanov) [inline Pod/Classes/LSPAudioViewController.m]: `nullable` annotation in `m` file.

File: Pod/Classes/LSPAudioViewController.m [hunk 6]
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
...
+{
+    __weak typeof(self) weakSelf = self;
+    [weakSelf.view layoutIfNeeded];
```
Comment (SergeyKuryanov) [inline Pod/Classes/LSPAudioViewController.m]: Why use `weakSelf` here?

File: Pod/Classes/LSPAudioViewController.m [hunk 7]
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
...
+    
+    NSTimeInterval duration = .666f;
+    NSTimeInterval delay = 0;
+    UIViewAnimationOptions options = (UIViewAnimationOptionBeginFromCurrentState | UIViewAnimationOptionCurveEaseInOut);
+    
+    [UIView animateWithDuration:duration delay:delay options:options animations:^{
+        __strong typeof(weakSelf) strongSelf = weakSelf;
```
Comment (SergeyKuryanov) [inline Pod/Classes/LSPAudioViewController.m]: Why use `strongSelf` here?

File: Pod/Classes/LSPAudioViewController.m [hunk 8]
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
...
+- (void)hide:(nullable void (^)(void))completion
+{
+    __weak typeof(self) weakSelf = self;
+    [weakSelf.view layoutIfNeeded];
+    
+    NSTimeInterval duration = .666f;
+    NSTimeInterval delay = 0;
+    UIViewAnimationOptions options = (UIViewAnimationOptionBeginFromCurrentState | UIViewAnimationOptionCurveEaseInOut);
+    
+    [UIView animateWithDuration:duration delay:delay options:options animations:^{
```
Comment (SergeyKuryanov) [inline Pod/Classes/LSPAudioViewController.m]: I think no reference cycles can appear in this block, so `weakify` is redundant here.

**Review by SergeyKuryanov — CHANGES_REQUESTED** [review]
Lots of useful changes. Nice work and congrats with version 1.0

For iPhone X Apple recommend extend bottom view's background to the end of the screen. Can be done in another PR.

Also when using example HarleQuin player, which have bigger height. Track title, playback bar and time labels are not centered. Is this expected?

File: Pod/Classes/LSPAudioPlayerModel.h (hunk 4 above)
Comment (adamyanalunas) [reply Pod/Classes/LSPAudioPlayerModel.h]: Xcode issue. Still belongs to the `loudspeaker` bundle. Will resolve.

File: Example/loudspeaker/LSPViewController.m (hunk 3 above)
Comment (adamyanalunas) [reply Example/loudspeaker/LSPViewController.m]: Good point. Will change.

File: Pod/Classes/LSPAudioViewController.m (hunk 6 above)
Comment (adamyanalunas) [reply Pod/Classes/LSPAudioViewController.m]: Figured if I needed to declare a `weakSelf` I may as well use it as long as I've got it. Will use `self.view` and declare the weak version closer to the block.

**Comment by adamyanalunas** [comment]
@SergeyKuryanov I believe I addressed all of your points. The layout does not degrade gracefully when using a non-standard height but I figured that could come in another PR. Maybe now that this only supports iOS 9 things could be changed to use `UIStackView` or the auto layout rules updated to be more flexible.

**Review by SergeyKuryanov — APPROVED** [review]
Looks good 👍

**Pull request merged**